database/src/main/java/database/service/JsonDatabaseService.java

```diff
@@ -145,6 +175,30 @@ public boolean removeRecordFromTable(Class<? extends BaseEntity> entityClass, In
         return true;
     }
 
+    @Override
+    public boolean removeSpecificRecords(Class<? extends BaseEntity> entityClass, List<Integer> ids) {
+        if (ids == null || ids.isEmpty()) {
+            throw new IllegalArgumentException("IDs list cannot be null or empty");
+        }
+
+        Path databasePath = Path.of(getDatabasePath(entityClass));
+
+        if (!Files.exists(databasePath)) {
+            LOG.error(DB_FILE_NOT_EXIST + ": {}", databasePath.toAbsolutePath());
+            throw new DatabaseDoesNotExistException(DB_FILE_NOT_EXIST);
```

Comment:
The same as for `addNewRecordsToTable`, consider moving validation into common method and use here and in `removeRecordFromTable` method.